pkg/build/controller/build/defaults/api/v1/types.go

```diff
@@ -43,6 +43,10 @@ type BuildDefaultsConfig struct {
 
 	// resources defines resource requirements to execute the build.
 	Resources kapi.ResourceRequirements `json:"resources,omitempty"`
+
+	// tolerations is a list of default Tolerations that are applied to build pods
+	// so that they can be placed on build-specific nodes
```

Comment:
needs good doc about what happens in the following cases:
* no tolerations on the pod, tolerations set here (assume these are all added verbatim)
* unrelated tolerations on the pod already, tolerations set here (assume these are appended)
* some of these toleration keys exist on the pod already, but with different values/effects (do these override or preserve what is already set on the pod?)

Reply:
@liggitt As of now, the only way that tolerations are set on build pods is through the defaulter, so there would be no other tolerations.  If/when there comes a time that users (or whoever) can add tolerations, then all of these things should be considered and addressed, but for now it might be overkill?
@bparees thoughts?

Reply:
for now it makes sense to state that these are appended to any existing tolerations on the build pod.

if there are multiple tolerations with the same key but different values, i would expect us to forgo adding the default toleration entry.

Reply:
(note that even the key is optional in a Toleration, so it's quite a mess deciding when two tolerations should replace each other vs both be added)

Reply:
@bparees updated in next push

Reply:
@bparees tolerations seem to be processed as a combination of the key/value/effect exactly equaling the taint, or the key/effect existing, and the value being blank, so it seems like skipping a toleration based on the key matching would not be the best option.

Reply:
@bparees updated description based on your feedback